build/gen_stub.php

```diff
@@ -2474,12 +2546,30 @@ function (Name $item) {
             $code .= $property->getDeclaration($allConstInfos);
         }
 
-        foreach ($this->attributes as $attribute) {
-            $code .= $attribute->generateCode("zend_add_class_attribute(class_entry", "class_$escapedName", $allConstInfos);
+        if (!empty($this->attributes)) {
+            if (!$php82MinimumCompatibility) {
+                $code .= "#if (PHP_VERSION_ID >= " . PHP_82_VERSION_ID . ")\n";
+            }
+
+            foreach ($this->attributes as $attribute) {
+                $code .= $attribute->generateCode("zend_add_class_attribute(class_entry", "class_$escapedName", $allConstInfos);
+            }
+
+            if (!$php82MinimumCompatibility) {
+                $code .= "#endif\n";
+            }
         }
 
         if ($attributeInitializationCode = generateAttributeInitialization($this->funcInfos, $this->cond)) {
+            if (!$php82MinimumCompatibility) {
+                $code .= "#if (PHP_VERSION_ID >= " . PHP_82_VERSION_ID . ")\n";
+            }
+
             $code .= "\n" . $attributeInitializationCode;
```

Comment:
**Minor**: CR will be better before first condition

For now, generated code is 
```
	class_entry = zend_register_internal_class_ex(&ce, NULL);
#if (PHP_VERSION_ID >= 80200)

#if defined(HAVE_MEMCACHED_SASL)
	zend_mark_function_parameter_as_sensitive(&class_entry->function_table, "setsaslauthdata", 1);
#endif
#endif

	return class_entry;

```
Will prefer 

```
	class_entry = zend_register_internal_class_ex(&ce, NULL);

#if (PHP_VERSION_ID >= 80200)
#if defined(HAVE_MEMCACHED_SASL)
	zend_mark_function_parameter_as_sensitive(&class_entry->function_table, "setsaslauthdata", 1);
#endif
#endif

	return class_entry;

```